### PR TITLE
Create an SDK PR guide

### PR DESCRIPTION
**Pull request by marcpopMSFT**

### CODE REVIEW

File: documentation/project-docs/SDK-PR-guide.md
Comment (akoeplinger) [inline documentation/project-docs/SDK-PR-guide.md:19]:
```suggestion
The .NET SDK has monthly servicing releases aligning with the .NET Runtime servicing releases. These are for top fixes and security updates only to limit risk.
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (akoeplinger) [inline documentation/project-docs/SDK-PR-guide.md:27]:
```suggestion
| Minor        | Quarterly    | Preview 3         | ~Prior release Preview 3 date | End of the month prior to Preview 3 (~7 weeks prior to release) |
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (akoeplinger) [inline documentation/project-docs/SDK-PR-guide.md:31]:
```suggestion
Even releases that are in lockdown can still take changes as long as they are approved and the final build isn't complete. To bring a change through tactics, mark it with the label servicing-consider and update the description to include 5 sections (Description, Customer Impact, Regression?, Risk, Testing). See previously approved bugs for examples by looking for the [servicing-approved](https://github.com/dotnet/sdk/pulls?q=is%3Apr+label%3AServicing-approved+is%3Aclosed) label
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (akoeplinger) [inline documentation/project-docs/SDK-PR-guide.md:36]:
```suggestion
- Ping `@dotnet/domestic-cat` when ready for a review
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (akoeplinger) [inline documentation/project-docs/SDK-PR-guide.md:44]:
```suggestion
`@dotnet/domestic-cat` will monitor codeflow and approve passing PRs (as all PRs require 1 approver) and ping the owning team for any failures.
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (akoeplinger) [inline documentation/project-docs/SDK-PR-guide.md:47]:
```suggestion
All monthly servicing releases are done of our internal branches in case there are security changes that have to go out. That means all flows from runtime branches into the SDK and from SDK to installer are done internally.
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (kartheekp-ms) [inline documentation/project-docs/SDK-PR-guide.md:21]: I assume we branch out or create a tag after code complete. For example, I see a tag created for December 2022 servicing https://github.com/dotnet/sdk/tree/v7.0.101.

I assume tags are created at the time of code complete. How about opening servicing branch for code flow without waiting for next branding PRs to be merged?

Comment (marcpopMSFT) [reply documentation/project-docs/SDK-PR-guide.md:21]: I think it's to prevent confusion where people merge, a build from the prior month gets created, and then someone tests it thinking it was last months build. I don't think there are specific engineering reasons other than avoiding confusion. @mmitche to confirm?

**Comment by nagilson** [comment]
Is this targeted more towards new hires and internal folks on other teams, or open source contributors?

**Comment by marcpopMSFT** [comment]
> Is this targeted more towards new hires and internal folks on other teams, or open source contributors?

@nagilson I think mostly internal partner teams. They are the most frequent source of questions about our schedule. Externals tend to not know enough of the inner details to even know what to ask.

**Comment by marcpopMSFT** [comment]
@dotnet/templating-engine-maintainers are you investigating the template engine leg failures as I've seen a lot of stability issues in these legs the past few days and this PR doesn't change code so can't break tests.

**Comment by MarcoRossignoli** [comment]
@marcpopMSFT thanks for the doc.

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:7]:
```suggestion
The .NET SDK has yearly major releases aligning with the .NET Runtime in November.  These releases will be of the form N.0.1xx. The feature work for major releases will be done in `main`. 
```
```suggestion
The .NET SDK has yearly major releases aligning with the .NET Runtime in November.  These releases will be of the form `N.0.1xx`. The feature work for major releases will be done in `main`. 

```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:9]:
```suggestion

During .NET 7, for example, the preview branches were created around the 21st of each month prior, with the later previews forking earlier to provide more time for stabilization and codeflow.
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:10]:
```suggestion
The `main` branch's major version will be set to match the next year's major release in August. (For example, the switch from `.NET 7` to `.NET 8` occurred in August of 2022, while .NET 8 was set for release in 2023.) Warning: This process can take a bit for those waiting to develop features for the next major release.
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:13]:
```suggestion
The .NET SDK has quarterly minor releases that align with Visual Studio releases. For example, two different quarterly releases in the same schedule would be: `7.0.100` vs `7.0.200`. These SDK releases are open to all bug fixes and some features work as well. 
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:14]:
```suggestion
These quarterly releases should avoid breaking changes if at all possible. Breaking changes instead should be included in the major releases and be tied to customer `TargetFramework` updates when possible. We do this to limit the pain of breaking changes, as customers need to take action (upgrading the `TargetFramework`) before they are broken. 
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:25]:
```suggestion
The .NET SDK has monthly servicing releases (e.g.`7.0.100` VS `7.0.101`) aligning with the .NET Runtime servicing releases. These are for top fixes and security updates only to limit risk.
```

Comment (nagilson) [reply documentation/project-docs/SDK-PR-guide.md:25]: Not sure if this suggestion is factually true... I don't recall seeing a `7.0.111`. for example.  Also, we may want to mention why there is no `7.1.00`, for example. This sometimes trips me up.

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:19]:
```suggestion
After the minor release locks down, it transitions into servicing / "QB" (Quarterback) mode. This typically happens around the `preview 3` release to allow time for any late public changes we want feedback on. The next release after `preview 3` is typically GA and done as a servicing release. The following steps are done when transitioning from minor release previews to servicing:
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:21]: We should clarify what stabilize branding is here.

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:22]:
```suggestion
- [ ] Enable internal runtime `darc` subscriptions and trigger them
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:26]:
```suggestion
Any servicing release is open for check-ins from the day the [branding PRs](https://github.com/dotnet/sdk/pulls?q=is%3Apr+branding) are merged (~1st of each month) and when code complete is (typically two weeks later).
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:3]:
```suggestion
## Branching & Release Schedules
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:5]: It may be worthwhile to mention the `alias` or places where internal folks can see the deadlines that tactics sends out. (I don't even think I get those emails still.)

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:37]:
```suggestion
Even releases that are in lockdown can still take changes as long as they are approved and the final build isn't complete. To bring a change through tactics, mark it with the label servicing-consider and update the description to include 5 sections (Description, Customer Impact, Regression?, Risk, Testing). See previously approved bugs for examples by looking for the [servicing-approved](https://github.com/dotnet/sdk/pulls?q=is%3Apr+label%3AServicing-approved+is%3Aclosed) label.
```

File: documentation/project-docs/SDK-PR-guide.md
Comment (nagilson) [inline documentation/project-docs/SDK-PR-guide.md:40]:
```suggestion
External contributions are encouraged and welcome. There are so many teams working in this repo that it can be hard to track. Contributors looking to learn more about how to contribute should check out the [Developer Guide](https://github.com/dotnet/sdk/blob/main/documentation/project-docs/developer-guide.md).
```

**Review by nagilson — COMMENTED** [review]
Marc, thanks for making this. It's going to make things a lot easier for contributors and new hires. Stuff like this is important but sometimes it gets pushed to the side for other work, which is unfortunate. Sorry it took a bit to review!

File: documentation/project-docs/SDK-PR-guide.md
Comment (marcpopMSFT) [reply documentation/project-docs/SDK-PR-guide.md:5]: I was trying to avoid internal links in the public document. I added some text that will direct people where to look for or ask.

**Review by nagilson — APPROVED** [review]

File: documentation/project-docs/SDK-PR-guide.md
Comment (marcpopMSFT) [reply documentation/project-docs/SDK-PR-guide.md]: This is accurate. We haven't gotten to 7.0.111 yet. We're on 6.0.114 now since it's been over a year since 6.0 shipped.

**Pull request merged**
